templates/repo/commits_list.tmpl

```diff
@@ -41,28 +41,18 @@
 									{{$class = (printf "%s%s" $class " isWarning")}}
 								{{end}}
 							{{end}}
+							{{$commitShaLink := ""}}
 							{{if $.PageIsWiki}}
-								<a href="{{$commitRepoLink}}/wiki/commit/{{.ID}}" rel="nofollow" class="{{$class}}">
+								{{$commitShaLink = (printf "%s/wiki/commit/%s" $commitRepoLink (PathEscape .ID.String))}}
 							{{else if $.PageIsPullCommits}}
-								<a href="{{$commitRepoLink}}/pulls/{{$.Issue.Index}}/commits/{{.ID}}" rel="nofollow" class="{{$class}}">
+								{{$commitShaLink = (printf "%s/pulls/%d/commits/%s" $commitRepoLink $.Issue.Index (PathEscape .ID.String))}}
 							{{else if $.Reponame}}
-								<a href="{{$commitRepoLink}}/commit/{{.ID}}" rel="nofollow" class="{{$class}}">
-							{{else}}
-								<span class="{{$class}}">
+								{{$commitShaLink = (printf "%s/commit/%s" $commitRepoLink (PathEscape .ID.String))}}
 							{{end}}
+							<a href="{{$commitShaLink}}" class="{{$class}}">
```

Comment:
There's a small difference here: Previously, the `else` block resulted in a `span`, now it results in an unclickable link.
But I doubt that'll be a problem.

Reply:
I am just going to change it to:

```suggestion
							<a {{if $commitShaLink}}href="{{$commitShaLink}}"{{end}} class="{{$class}}">
```

🤣 

A `<a>` without `href` is like a span 😁